DecoratorDesignPattern/Decorator/ExtraCheeseTopping.java

```diff
@@ -0,0 +1,11 @@
+public class ExtraCheeseTopping extends ToppingDecorator {
+    BasePizza pizza;
+    ExtraCheeseTopping(BasePizza pizza) {
+        this.pizza = pizza;
+    }
+
```

Comment:
we can remove code duplication and code readability by

moving this logic inside TopicDecorator and use as -

`ExtraCheeseTopping(BasePizza pizza) {
        super(pizza)
    }`